### PR TITLE
Added Xcode version check and logging utilities for kernel errors

### PR DESCRIPTION
**Pull request by petewarden**

### CODE REVIEW

**Comment by petewarden** [comment]
@ebrevdo Can you take a look at the LogAllRegisteredKernels() function I added to tensorflow/core/framework/op_kernel.cc? I think it's trivial, but it's pretty useful for debugging missing kernels on mobile builds where we strip a lot out.

**Comment by petewarden** [comment]
Jenkins, test this please.

File: tensorflow/core/framework/op_kernel.cc [hunk 1]
```diff
@@ -774,6 +774,13 @@ Status SupportedDeviceTypesForNode(
   return Status::OK();
 }
 
+void LogAllRegisteredKernels() {
+  for (const auto& key_registration : *GlobalKernelRegistryTyped()) {
+    const KernelDef& kernel_def(key_registration.second.def);
```
Comment (ebrevdo) [inline tensorflow/core/framework/op_kernel.cc]: consider adding a prefix & suffix LOG and indent the LOG(INFO)  up to you.

**Comment by ebrevdo** [comment]
Failing tests with python3?

**Comment by ebrevdo** [comment]
Jenkins, test this please.

**Comment by rmlarsen** [comment]
@ebrevdo is this ready to be merged?

**Comment by petewarden** [comment]
Thanks for the review! Since the changes were fairly small, but I have a high-priority iOS fix to get in, I went ahead and merged.

**Pull request merged**